core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java

```diff
@@ -153,6 +157,9 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
                 } else {
                     LOGGER.warn("Unhandled stack status, {}", stackStatus);
                 }
+                if (stackStatus != null) {
+                    metricsClient.processStackStatus(stack.getResourceCrn(), stack.cloudPlatform(), stackStatus.name(), stackStatus.ordinal());
+                }
```

Comment:
do you have any measurement about the length of this call? I would guess this should be very fast/async/configured with small timeout in order to not block syncer

Reply:
only the entitlement check which is sync, any other operations are happening on a different thread

Reply:
so everytime we will open 1 extra thread when a sync happens. it still can cause issue if those threads are not closed in short time, can't it?

Reply:
it's the same thread every time, depends on the number of workers, basically we just put data to a queue (with limited size) and a worker from a different thread will process that (or more workers), you can check the implementation here: https://github.com/hortonworks/cloudbreak/tree/master/streaming-common/src/main/java/com/sequenceiq/cloudbreak/streaming/processor